core/trino-main/src/main/java/io/trino/cost/CboFeaturesConfig.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cost;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.LegacyConfig;
+import io.airlift.units.DataSize;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static java.util.Objects.requireNonNull;
+
+public class CboFeaturesConfig
```

Comment:
CBO is not "a thing", so this config gets injected into various places. from code maintenance perspective, does this separation buy us anything?

Note that the commit "Extract CBO configs into CboFeaturesConfig" currently doesn't carry any rationale for the change.

Reply:
> Note that the commit "Extract CBO configs into CboFeaturesConfig" currently doesn't carry any rationale for the change.

`FeaturesConfig` became a dump ground (https://en.wikipedia.org/wiki/God_object) for properties that don't have better place to go. IMO we are above property count threshold for CBO properties where it makes sense to group them together as separate config class (similar for session properties). Otherwise is just becomes messier and messier

Reply:
Can you please add this nice rationale to the commit message?

Reply:
I've updated commit message now